ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureCloudDestinationSync.java

```diff
@@ -714,27 +727,46 @@ protected int eraseBlobs(List<BlobItem> blobItemList, BlobContainerClient blobCo
         long expirationTime = Long.parseLong(metadata.get(CloudBlobMetadata.FIELD_EXPIRATION_TIME));
         eraseBlob = (expirationTime + gracePeriod) < now;
         eraseReason = String.format("%s: (%s + %s) < %s", CloudBlobMetadata.FIELD_EXPIRATION_TIME, expirationTime, gracePeriod, now);
+      } else if (deletedContainers.contains(shortShortPair)) {
+        eraseBlob = true;
+        eraseReason = String.format("account = %s, deleted_container = %s", shortShortPair.getFirst(), shortShortPair.getSecond());
+      } else {
+        // nothing to do, blob cannot be deleted
+        logger.trace("[COMPACT] No reason to erase blob {} from Azure blob storage", blobItem.getName());
       }
 
       if (eraseBlob) {
         if (cloudConfig.cloudCompactionDryRunEnabled) {
-          logger.trace("[DRY-RUN] Can erase blob {} from Azure blob storage because {}", blobItem.getName(), eraseReason);
+          logger.trace("[DRY-RUN][COMPACT] Can erase blob {} from Azure blob storage because {}", blobItem.getName(), eraseReason);
         } else {
           Timer.Context storageTimer = azureMetrics.blobCompactionLatency.time();
-          logger.trace("Erasing blob {} from Azure blob storage because {}", blobItem.getName(), eraseReason);
+          logger.trace("[COMPACT] Erasing blob {} from Azure blob storage because {}", blobItem.getName(), eraseReason);
           blobContainerClient.getBlobClient(blobItem.getName()).delete();
           storageTimer.stop();
           vcrMetrics.blobCompactionRate.mark();
         }
         numBlobsPurged += 1;
-      } else {
-        logger.trace("Cannot erase blob {} from Azure blob storage because condition not met: {}", blobItem.getName(), eraseReason);
+      } else if (eraseReason != null) {
```

Comment:
the if condition is  confusing.
Assume 
1. if eraseBlob is true,  eraseReason for sure is not null
2. if eraseBlob is false, eraseReason is null.
Is there any case, eraseBlob is false but eraseReason is not null?

Reply:
simplified it. hope its readable now.